tests/test_devices/test_okdev.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+from pathlib import Path
+from mio.devices.mocks import okDevMock
+
+
+@pytest.fixture
+def _tmp_buffer(tmp_path: Path, set_okdev_input) -> bytes:
+    buffer = b"12345"
+    buffer_rep = b"12345" * 3
+    data_path = tmp_path / "data.bin"
+    with open(data_path, "wb") as f:
+        f.write(buffer_rep)
+    set_okdev_input(data_path)
+    return buffer
+
+
+def test_okdev_iter(_tmp_buffer: bytes) -> None:
+    """
+    sanity check that the __iter__ method for iteration on the okdev
+    does iter things
+    """
+    dev = okDevMock(read_length=5)
+    got_buffers = 0
+    for buf in dev:
+        assert buf == _tmp_buffer
+        got_buffers += 1
+    assert got_buffers == 3
+
+
+def test_okdev_next(_tmp_buffer: bytes) -> None:
+    """
+    the okDevMock.__next__ method (and by proxy okDev) does next things
+    """
+    dev = okDevMock(read_length=5)
+    buffers = []
+    buffers.append(next(dev))
+    buffers.append(next(dev))
+    buffers.append(next(dev))
+    with pytest.raises(StopIteration):
```

Comment:
i don't think i explained this in the video call - @hsemwal what this means is that the test fails if the error is *not* raised here